docs/core/tutorials/creating-app-with-plugin-support.md

```diff
@@ -0,0 +1,319 @@
+
+## Create a plugin with library dependencies
+
+Almost all plugins are more complex than a simple "Hello World", and many plugins have dependencies on other libraries. The `JsonPlugin` and `OldJson` plugin projects in the sample show two examples of plugins with NuGet package dependencies on `Newtonsoft.Json`. The project files themselves do not have any special information for the project references, and (after adding the plugin paths to the `pluginPaths` array) the plugins run perfectly, even if run in the same execution of the AppWithPlugin app. However, these projects do not copy the referenced assemblies to their output directory, so the assemblies need to be present on the user's machine for the plugins to work. Since that isn't a good user experience, add the `<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>`  property in between the `<PropertyGroup>` tags in the plugin's project file. See the `XcopyablePlugin` plugin project for an example.
+
+## Other plugin examples in the sample
+
+The `AssemblyDependencyResolver` object can also resolve native libraries included in NuGet packages as well as localized satellite assemblies. The `UVPlugin` and `FrenchPlugin` demonstrate these scenarios respectively.
+
+## How to reference a plugin interface assembly defined in a NuGet package
+
+Let's say that there is an app A that has a plugin interface defined in the NuGet package named `A.PluginBase`. How do you reference the package correctly in your plugin project? For project references, using the `<Private>false</Private>` metadata on the `ProjectReference` element in the project file prevented the dll from being copied to the output.
+
+To correctly reference the `A.PluginBase` package, you want to change the `<PackageReference>` element in the project file to the following:
+
+```xml
+<PackageReference Include="A.PluginBase" Version="1.0.0">
+    <ExcludeAssets>runtime</ExcludeAssets>
+</PackageReference>
+```
+
+This will prevent the `A.PluginBase` assemblies from being copied to the output directory of your plugin and ensure that your plugin will use A's version of `A.PluginBase`.
+
+## Plugin target framework recommendations
+
+Because plugin dependency loading uses the `deps.json` file, there are some gotchas with it. Specifically, your plugins should target a runtime such as .NET Core 3.0 instead of a version of .NET Standard. The `deps.json` file is generated based on which framework the project is targeting, and since many .NET Standard-compatible packages ship reference assemblies for building against .NET Standard and implementation assemblies for specific runtimes, the `deps.json` may incorrectly not see implementation assemblies or grab the .NET Standard version of an assembly instead of the .NET Core version you may be expecting.
```

Comment:
```suggestion
Because plugin dependency loading uses the *deps.json* file, there are some gotchas with it. Specifically, your plugins should target a runtime, such as .NET Core 3.0, instead of a version of .NET Standard. The `deps.json` file is generated based on which framework the project targets, and since many .NET Standard-compatible packages ship reference assemblies for building against .NET Standard and implementation assemblies for specific runtimes, the `deps.json` may not correctly see implementation assemblies, or it may grab the .NET Standard version of an assembly instead of the .NET Core version you expect.
```

Reply:
The "some gotchas", though, doesn't correspond to the one gotcha that you list.